src/dune/build_system.ml

```diff
@@ -1368,6 +1422,16 @@ end = struct
       force_rerun || depends_on_universe
     in
     let rule_digest = compute_rule_digest rule ~deps ~action ~sandbox_mode in
+    let () =
+      if Action.is_useful_to_distribute action = Maybe then
+        let f { cache = (module Caching); _ } =
+          match Caching.Cache.hint Caching.cache [ rule_digest ] with
+          | Result.Ok _ -> ()
+          | Result.Error e ->
+            User_warning.emit [ Pp.textf "unable to hint the cache: %s" e ]
```

Comment:
Do we really want to flood the used with warnings if the cache is down?

Reply:
I don't really mind. I think the user should be warned, but maybe just once ? @jeremiedimino ?

Reply:
Well, if we're building thousands of rules, which is typical, thousands of warnings would make the Dune hard to use.

Reply:
Hence maybe just once, but it would IMO be a mistake to entirely silence such warnings. If the user enables the cache, but the cache does not work for some reason, something should be said, lest we will never be sure if things are actually getting cached.

Reply:
Yep, I think making this warning just once may be a good compromise. The text of the warning should let the user know that subsequent similar warnings would be silenced.